src/List.mo

```diff
@@ -1165,21 +1335,24 @@ module {
       block
     };
 
+    var i = 1;
+    var pos = 0;
+
     while (i < blockIndex) {
-      let len = data_block_size(i);
-      data_blocks[i] := make_block(len, len);
+      let len = dataBlockSize(i);
+      dataBlocks[i] := makeBlock(array, pos, len, len);
+      pos += len;
       i += 1
     };
     if (elementIndex != 0 and blockIndex < blocks) {
-      data_blocks[i] := make_block(data_block_size(i), elementIndex)
+      dataBlocks[i] := makeBlock(array, pos, dataBlockSize(i), elementIndex)
```

Comment:
I think the condition blockIndex < blocks in the line above is redundant. if elementIndex > 0 then blocks = newIndexBlockLength(blockIndex) > blockIndex always. I think we should remove the condition and just say if (elementIndex > 0) {